specifyweb/businessrules/interaction_rules.py

```diff
@@ -34,20 +34,39 @@ def loanprep_quantity_must_be_lte_availability(ipreparation):
         quantity = ipreparation.quantity or 0
         quantityresolved = ipreparation.quantityresolved or 0
         if available < (quantity - quantityresolved):
-            raise BusinessRuleException(f"loan preparation quantity exceeds availability ({ipreparation.id}: {quantity - quantityresolved} {available})")
+            raise BusinessRuleException(
```

Comment:
A big ask, but can you check if front-end already has all of these business rules?

Handling business rules on the front-end has benefits:
- Faster error feedback
- We know exactly which field the error related to

Of course, we still need to have back-end business rules for direct API calls and for workbench